dynamiqs/mesolve/merouchon.py

```diff
@@ -0,0 +1,80 @@
+# ruff: noqa: ANN001, ANN201, ARG002
+# we mostly ignore type hinting in this file for readability purposes
+
+from __future__ import annotations
+
+import diffrax as dx
+import jax.numpy as jnp
+from diffrax._custom_types import RealScalarLike, Y
+from diffrax._local_interpolation import LocalLinearInterpolation
+from jax import Array
+
+from ..core.abstract_solver import MESolver
+from ..core.diffrax_solver import FixedSolver
+from ..utils.utils.general import dag
+
+
+class AbstractRouchonTerm(dx.AbstractTerm):
+    # this class bypasses the typical Diffrax term implementation, as Rouchon schemes
+    # don't match the vf/contr/prod structure
+
+    kraus_map: callable[[RealScalarLike, RealScalarLike, Y], Y]
+    # should be defined as `kraus_map(t0, t1, y0) -> y1`
+
+    def vf(self, t, y, args):
+        pass
+
+    def contr(self, t0, t1, **kwargs):
+        pass
+
+    def prod(self, vf, control):
+        pass
+
+
+class RouchonSolver(dx.AbstractSolver):
+    term_structure = AbstractRouchonTerm
+    interpolation_cls = LocalLinearInterpolation
+
+    def init(self, terms, t0, t1, y0, args):
+        pass
+
+    def order(self, terms):
+        return 1
+
+    def step(self, terms, t0, t1, y0, args, solver_state, made_jump):
+        del solver_state, made_jump
```

Comment:
```suggestion
        del solver_state, made_jump, args
```
(unsure about this one, but it's not used either)

Reply:
We could do this but it's useless as far as I understand (args is an empty tuple, we won't gain anything by deleting it).